source/server/hot_restarting_base.h

```diff
@@ -91,13 +91,15 @@ class RpcStream : public Logger::Loggable<Logger::Id::main> {
  */
 class HotRestartingBase : public Logger::Loggable<Logger::Id::main> {
 protected:
-  HotRestartingBase(uint64_t base_id) : main_rpc_stream_(base_id) {}
+  HotRestartingBase(uint64_t base_id)
+      : main_rpc_stream_(base_id), udp_forwarding_rpc_stream_(base_id) {}
 
   // Returns a Gauge that tracks hot-restart generation, where every successive
   // child increments this number.
   static Stats::Gauge& hotRestartGeneration(Stats::Scope& scope);
 
   RpcStream main_rpc_stream_;
+  RpcStream udp_forwarding_rpc_stream_;
```

Comment:
It's a separate stream because udp packets can be forwarded at any time after the drain begins, and main_rpc_stream_ communication is still potentially happening at that point. Single threading of the HotRestarter should be enough to prevent individual messages from comingling, but there's nothing stopping parent instance from sending a UDP packet while child instance is sending a request for which it will expect to read a response. A distinct channel prevents any conflicts of that nature.

Reply:
Makes sense. Can you add a comment to that effect somewhere?

Reply:
Commented both the RpcStreams.